resteasy-client/src/main/java/org/jboss/resteasy/plugins/providers/sse/client/SseEventSourceImpl.java

```diff
@@ -367,10 +435,13 @@ public void run()
          }
          catch (Throwable e)
          {
+            waitForSubscriber();
```

Comment:
@ronsigal, Do we need to notify client error immediately when there is error happens instead of wait for the readLatch to countdown?